README.md

```diff
@@ -35,6 +35,27 @@ $ docker build -t my-nodejs-app .
 $ docker run -it --rm --name my-running-app my-nodejs-app
 ```
 
+If you prefer Docker Compose:
+
+```yaml
+version: "2"
+services:
+  node:
+    image: "node:4-onbuild"
```

Comment:
Nope, it does not work that way.  `docker-compose` will only build an image if there is a `build:` key and `ONBUILD` directives in a Dockerfile only get fired in a dependent image build.

I took exactly your four files and it fails because most likely you have `express` installed in your local directory which you then mount in the container with `volumes:` in the yaml.

```console
$ docker-compose up
Creating network "node_default" with the default driver
Creating node_node_1
Attaching to node_node_1
node_1  | npm info it worked if it ends with ok
node_1  | npm info using npm@2.15.11
node_1  | npm info using node@v4.7.3
node_1  | npm info prestart docker-test@1.0.0
node_1  | npm info start docker-test@1.0.0
node_1  | 
node_1  | > docker-test@1.0.0 start /usr/src/app
node_1  | > node index.js
node_1  | 
node_1  | module.js:327
node_1  |     throw err;
node_1  |     ^
node_1  | 
node_1  | Error: Cannot find module 'express'
node_1  |     at Function.Module._resolveFilename (module.js:325:15)
node_1  |     at Function.Module._load (module.js:276:25)
node_1  |     at Module.require (module.js:353:17)
node_1  |     at require (internal/module.js:12:17)
node_1  |     at Object.<anonymous> (/usr/src/app/index.js:1:77)
node_1  |     at Module._compile (module.js:409:26)
node_1  |     at Object.Module._extensions..js (module.js:416:10)
node_1  |     at Module.load (module.js:343:32)
node_1  |     at Function.Module._load (module.js:300:12)
node_1  |     at Function.Module.runMain (module.js:441:10)
node_1  | 
node_1  | npm info docker-test@1.0.0 Failed to exec start script
node_1  | npm ERR! Linux 4.9.3-gentoo
node_1  | npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
node_1  | npm ERR! node v4.7.3
node_1  | npm ERR! npm  v2.15.11
node_1  | npm ERR! code ELIFECYCLE
node_1  | npm ERR! docker-test@1.0.0 start: `node index.js`
node_1  | npm ERR! Exit status 1
node_1  | npm ERR! 
node_1  | npm ERR! Failed at the docker-test@1.0.0 start script 'node index.js'.
node_1  | npm ERR! This is most likely a problem with the docker-test package,
node_1  | npm ERR! not with npm itself.
node_1  | npm ERR! Tell the author that this fails on your system:
node_1  | npm ERR!     node index.js
node_1  | npm ERR! You can get information on how to open an issue for this project with:
node_1  | npm ERR!     npm bugs docker-test
node_1  | npm ERR! Or if that isn't available, you can get their info via:
node_1  | npm ERR! 
node_1  | npm ERR!     npm owner ls docker-test
node_1  | npm ERR! There is likely additional logging output above.
node_1  | 
node_1  | npm ERR! Please include the following file with any support request:
node_1  | npm ERR!     /usr/src/app/npm-debug.log
node_node_1 exited with code 1
```